lib/rules/color-named/index.js

```diff
@@ -58,11 +58,23 @@ const rule = function (expectation, options) {
         return
       }
 
+      const functionNodeIndexes = []
+
       valueParser(decl.value).walk(node => {
         const value = node.value,
           type = node.type,
           sourceIndex = node.sourceIndex
 
+        // Find nodes that are inside functions
+        if (type === "function") {
+          node.nodes.forEach((node) => functionNodeIndexes.push(node.sourceIndex))
+        }
+
+        // Return early if node is inside of a function
+        if (functionNodeIndexes.indexOf(sourceIndex) > -1) {
```

Comment:
Additionally, it appears returning `false` inside of a function node is all that needs to be done.